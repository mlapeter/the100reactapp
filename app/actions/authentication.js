export const FETCH_TOKEN = "FETCH_TOKEN";
export const FETCH_TOKEN_RESULT = "FETCH_TOKEN_RESULT";
export const FETCH_TOKEN_ERROR = "FETCH_TOKEN_ERROR";

export const DECODE_TOKEN = "DECODE_TOKEN";
export const DECODE_TOKEN_RESULT = "DECODE_TOKEN_RESULT";
export const DECODE_TOKEN_ERROR = "DECODE_TOKEN_ERROR";
export const REMOVE_TOKEN = "REMOVE_TOKEN";
export const REMOVE_TOKEN_ERROR = "REMOVE_TOKEN_ERROR";
export const SET_FIREBASE_TOKEN = "SET_FIREBASE_TOKEN";
export const RESET_PASSWORD = "RESET_PASSWORD";
export const RESET_PASSWORD_RESULT = "RESET_PASSWORD_RESULT";
export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR";

export const fetchToken = (username, password, temp_auth_token) => ({
  type: FETCH_TOKEN,
  username,
  password,
  temp_auth_token
});

export const decodeToken = token => ({
  type: DECODE_TOKEN,
  token
});

export const removeToken = () => ({
  type: REMOVE_TOKEN
});

export const setFirebaseToken = token => ({
  type: SET_FIREBASE_TOKEN,
  token
});

export const resetPassword = resetPasswordEmail => ({
  type: RESET_PASSWORD,
  resetPasswordEmail
});
