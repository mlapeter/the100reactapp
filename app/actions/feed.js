export const FETCH_FEED = "FETCH_FEED";
export const FETCH_FEED_RESULT = "FETCH_FEED_RESULT";
export const FETCH_FEED_ERROR = "FETCH_FEED_ERROR";
export const FETCH_FEED_NO_DATA = "FETCH_FEED_NO_DATA";
export const LOAD_MORE_FEED_ITEMS = "LOAD_MORE_FEED_ITEMS";
export const LOAD_MORE_FEED_ITEMS_RESULT = "LOAD_MORE_FEED_ITEMS_RESULT";
export const CHANGE_FEED_PAGE = "CHANGE_FEED_PAGE";

export const fetchFeed = () => ({
  type: FETCH_FEED
});

export const loadMoreFeedItems = () => ({
  type: LOAD_MORE_FEED_ITEMS
});
