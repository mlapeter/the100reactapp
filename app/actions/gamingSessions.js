export const CREATE_GAMING_SESSION = "CREATE_GAMING_SESSION";
export const CREATE_GAMING_SESSION_RESULT =
  "CREATE_GAMING_SESSION_ERROR_RESULT";
export const CREATE_GAMING_SESSION_ERROR = "CREATE_GAMING_SESSION_ERROR";

export const EDIT_GAMING_SESSION = "EDIT_GAMING_SESSION";
export const EDIT_GAMING_SESSION_RESULT = "EDIT_GAMING_SESSION_RESULT";
export const EDIT_GAMING_SESSION_ERROR = "EDIT_GAMING_SESSION_ERROR";

export const DELETE_GAMING_SESSION = "DELETE_GAMING_SESSION";
export const DELETE_GAMING_SESSION_RESULT = "DELETE_GAMING_SESSION_RESULT";
export const DELETE_GAMING_SESSION_ERROR = "DELETE_GAMING_SESSION_ERROR";

export const FETCH_GAMING_SESSION = "FETCH_GAMING_SESSION";
export const FETCH_GAMING_SESSION_RESULT = "FETCH_GAMING_SESSION_RESULT";
export const FETCH_GAMING_SESSION_ERROR = "FETCH_GAMING_SESSION_ERROR";

export const FETCH_GAMING_SESSIONS = "FETCH_GAMING_SESSIONS";
export const FETCH_GAMING_SESSIONS_RESULT = "FETCH_GAMING_SESSIONS_RESULT";
export const FETCH_GAMING_SESSIONS_ERROR = "FETCH_GAMING_SESSIONS_ERROR";
export const FETCH_GAMING_SESSIONS_NO_DATA = "FETCH_GAMING_SESSIONS_NO_DATA";
export const REFRESH_GAMING_SESSIONS = "REFRESH_GAMING_SESSIONS";
export const LOAD_MORE_GAMING_SESSIONS = "LOAD_MORE_GAMING_SESSIONS";
export const LOAD_MORE_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_GAMING_SESSIONS_RESULT";

export const FETCH_MY_GAMING_SESSIONS = "FETCH_MY_GAMING_SESSIONS";
export const FETCH_MY_GAMING_SESSIONS_RESULT =
  "FETCH_MY_GAMING_SESSIONS_RESULT";
export const FETCH_MY_GAMING_SESSIONS_ERROR = "FETCH_MY_GAMING_SESSIONS_ERROR";
export const FETCH_MY_GAMING_SESSIONS_NO_DATA =
  "FETCH_MY_GAMING_SESSIONS_NO_DATA";
export const CLEAR_MY_GAMING_SESSIONS = "ClEAR_MY_GAMING_SESSIONS";
export const REFRESH_MY_GAMING_SESSIONS = "REFRESH_MY_GAMING_SESSIONS";
export const LOAD_MORE_MY_GAMING_SESSIONS = "LOAD_MORE_MY_GAMING_SESSIONS";
export const LOAD_MORE_MY_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_MY_GAMING_SESSIONS_RESULT";

export const FETCH_GROUP_GAMING_SESSIONS = "FETCH_GROUP_GAMING_SESSIONS";
export const FETCH_GROUP_GAMING_SESSIONS_RESULT =
  "FETCH_GROUP_GAMING_SESSIONS_RESULT";
export const FETCH_GROUP_GAMING_SESSIONS_ERROR =
  "FETCH_GROUP_GAMING_SESSIONS_ERROR";
export const FETCH_GROUP_GAMING_SESSIONS_NO_DATA =
  "FETCH_GROUP_GAMING_SESSIONS_NO_DATA";

export const CLEAR_GROUP_GAMING_SESSIONS = "CLEAR_GROUP_GAMING_SESSIONS";

export const REFRESH_GROUP_GAMING_SESSIONS = "REFRESH_GROUP_GAMING_SESSIONS";
export const LOAD_MORE_GROUP_GAMING_SESSIONS =
  "LOAD_MORE_GROUP_GAMING_SESSIONS";
export const LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT";

export const createGamingSession = gamingSession => ({
  type: CREATE_GAMING_SESSION,
  gamingSession
});

export const editGamingSession = gamingSession => ({
  type: EDIT_GAMING_SESSION,
  gamingSession
});

export const deleteGamingSession = deleteGamingSessionId => ({
  type: DELETE_GAMING_SESSION,
  deleteGamingSessionId
});

export const fetchGamingSession = gamingSessionId => ({
  type: FETCH_GAMING_SESSION,
  gamingSessionId
});

export const fetchGamingSessions = endpoint => ({
  type: FETCH_GAMING_SESSIONS,
  endpoint
});

export const refreshGamingSessions = endpoint => ({
  type: REFRESH_GAMING_SESSIONS,
  endpoint
});

export const loadMoreGamingSessions = endpoint => ({
  type: LOAD_MORE_GAMING_SESSIONS,
  endpoint
});

export const fetchMyGamingSessions = endpoint => ({
  type: FETCH_MY_GAMING_SESSIONS,
  endpoint
});

export const refreshMyGamingSessions = endpoint => ({
  type: REFRESH_MY_GAMING_SESSIONS,
  endpoint
});

export const loadMoreMyGamingSessions = endpoint => ({
  type: LOAD_MORE_MY_GAMING_SESSIONS,
  endpoint
});

export const fetchGroupGamingSessions = endpoint => ({
  type: FETCH_GROUP_GAMING_SESSIONS,
  endpoint
});

export const refreshGroupGamingSessions = endpoint => ({
  type: REFRESH_GROUP_GAMING_SESSIONS,
  endpoint
});

export const loadMoreGroupGamingSessions = endpoint => ({
  type: LOAD_MORE_GROUP_GAMING_SESSIONS,
  endpoint
});
