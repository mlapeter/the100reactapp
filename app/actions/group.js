export const FETCH_GROUP = "FETCH_GROUP";
export const FETCH_GROUP_RESULT = "FETCH_GROUP_RESULT";
export const FETCH_GROUP_ERROR = "FETCH_GROUP_ERROR";
export const FETCH_GROUP_EMPTY = "FETCH_GROUP_EMPTY";

export const CHANGE_GROUP = "CHANGE_GROUP";

export const fetchGroup = groupId => ({
  type: FETCH_GROUP,
  groupId
});

export const changeSelectedGroupId = groupId => ({
  type: CHANGE_GROUP,
  groupId
});
