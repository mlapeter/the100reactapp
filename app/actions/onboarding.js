export const SET_PLATFORM = "SET_PLATFORM";
export const SET_GAMERTAG = "SET_GAMERTAG";
export const SET_GAME = "SET_GAME";
export const SET_PROFILE_INFO = "SET_PROFILE_INFO";
export const CREATE_USER = "CREATE_USER";
export const CREATE_USER_ERROR = "CREATE_USER_ERROR";

export const setPlatform = platform => ({
  type: SET_PLATFORM,
  platform
});

export const setGamertag = gamertag => ({
  type: SET_GAMERTAG,
  gamertag
});

export const setGame = gameId => ({
  type: SET_GAME,
  gameId
});

export const setProfileInfo = (
  timezone,
  playStyle,
  age,
  playSchedule,
  group
) => ({
  type: SET_PROFILE_INFO,
  timezone,
  playStyle,
  age,
  playSchedule,
  group
});
export const createUser = (email, password, tos_privacy_agreement) => ({
  type: CREATE_USER,
  email,
  password,
  tos_privacy_agreement
});
