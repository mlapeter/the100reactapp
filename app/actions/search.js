export const CHANGE_ACTIVITY = "CHANGE_ACTIVITY";
export const CHANGE_GAME = "CHANGE_GAME";
export const CHANGE_GAMING_SESSIONS_PAGE = "CHANGE_GAMING_SESSIONS_PAGE";
export const CHANGE_MY_GAMING_SESSIONS_PAGE = "CHANGE_MY_GAMING_SESSIONS_PAGE";
export const CHANGE_GROUP_GAMING_SESSIONS_PAGE =
  "CHANGE_GROUP_GAMING_SESSIONS_PAGE";
export const CHANGE_RECENT_GAMING_SESSIONS_PAGE =
  "CHANGE_RECENT_GAMING_SESSIONS_PAGE";
export const CHANGE_PLATFORM = "CHANGE_PLATFORM";
export const TOGGLE_NOT_FULL = "TOGGLE_NOT_FULL";
export const FETCH_GAMES = "FETCH_GAMES";
export const FETCH_GAMES_RESULT = "FETCH_GAMES_RESULT";
export const FETCH_GAMES_ERROR = "FETCH_GAMES_ERROR";
export const FETCH_ACTIVITIES_RESULT = "FETCH_ACTIVITIES_RESULT";

export const changeActivity = activity => ({
  type: CHANGE_ACTIVITY,
  activity
});

export const changeGame = gameId => ({
  type: CHANGE_GAME,
  gameId
});

export const changePlatform = platform => ({
  type: CHANGE_PLATFORM,
  platform
});

export const fetchGames = games => ({
  type: FETCH_GAMES,
  games
});

export const toggleNotFull = notFull => ({
  type: TOGGLE_NOT_FULL,
  notFull
});
