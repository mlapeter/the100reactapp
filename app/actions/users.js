export const UPDATE_USER = "UPDATE_USER";
export const UPDATE_USER_RESULT = "UPDATE_USER_RESULT";
export const UPDATE_USER_ERROR = "UPDATE_USER_ERROR";

export const FETCH_USER = "FETCH_USER";
export const FETCH_USER_RESULT = "FETCH_USER_RESULT";
export const FETCH_USER_ERROR = "FETCH_USER_ERROR";

export const FETCH_CURRENT_USER = "FETCH_CURRENT_USER";
export const FETCH_CURRENT_USER_RESULT = "FETCH_CURRENT_USER_RESULT";
export const CLEAR_CURRENT_USER = "CLEAR_CURRENT_USER";

export const FETCH_FRIENDS = "FETCH_FRIENDS";
export const FETCH_FRIENDS_RESULT = "FETCH_FRIENDS_RESULT";
export const FETCH_FRIENDS_ERROR = "FETCH_FRIENDS_ERROR";
export const FETCH_FRIENDS_NO_DATA = "FETCH_FRIENDS_NO_DATA";
export const LOAD_MORE_FRIENDS = "LOAD_MORE_FRIENDS";
export const LOAD_MORE_FRIENDS_RESULT = "LOAD_MORE_FRIENDS_RESULT";
export const CHANGE_FRIENDS_PAGE = "CHANGE_FRIENDS_PAGE";
export const REFRESH_FRIENDS = "REFRESH_FRIENDS";

export const FETCH_PENDING_FRIENDS = "FETCH_PENDING_FRIENDS";
export const FETCH_PENDING_FRIENDS_RESULT = "FETCH_PENDING_FRIENDS_RESULT";
export const FETCH_PENDING_FRIENDS_ERROR = "FETCH_PENDING_FRIENDS_ERROR";
export const FETCH_PENDING_FRIENDS_NO_DATA = "FETCH_PENDING_FRIENDS_NO_DATA";
export const LOAD_MORE_PENDING_FRIENDS = "LOAD_MORE_PENDING_FRIENDS";
export const LOAD_MORE_PENDING_FRIENDS_RESULT =
  "LOAD_MORE_PENDING_FRIENDS_RESULT";

export const CHANGE_PENDING_FRIENDS_PAGE = "CHANGE_PENDING_FRIENDS_PAGE";
export const REFRESH_PENDING_FRIENDS = "REFRESH_PENDING_FRIENDS";

export const FETCH_GROUP_MEMBERS = "FETCH_GROUP_MEMBERS";
export const FETCH_GROUP_MEMBERS_RESULT = "FETCH_GROUP_MEMBERS_RESULT";
export const FETCH_GROUP_MEMBERS_ERROR = "FETCH_GROUP_MEMBERS_ERROR";
export const FETCH_GROUP_MEMBERS_NO_DATA = "FETCH_GROUP_MEMBERS_NO_DATA";
export const LOAD_MORE_GROUP_MEMBERS = "LOAD_MORE_GROUP_MEMBERS";
export const LOAD_MORE_GROUP_MEMBERS_RESULT = "LOAD_MORE_GROUP_MEMBERS_RESULT";
export const CHANGE_GROUP_MEMBERS_PAGE = "CHANGE_GROUP_MEMBERS_PAGE";
export const REFRESH_GROUP_MEMBERS = "REFRESH_GROUP_MEMBERS";

export const fetchUser = userId => ({
  type: FETCH_USER,
  userId
});

export const fetchCurrentUser = () => ({
  type: FETCH_CURRENT_USER
});

export const updateUser = currentUser => ({
  type: UPDATE_USER,
  currentUser
});

export const clearCurrentUser = () => ({
  type: CLEAR_CURRENT_USER
});

export const fetchFriends = () => ({
  type: FETCH_FRIENDS
});

export const loadMoreFriends = () => ({
  type: LOAD_MORE_FRIENDS
});

export const changeFriendsPage = () => ({
  type: CHANGE_FRIENDS_PAGE
});

export const refreshFriends = () => ({
  type: REFRESH_FRIENDS
});

export const fetchPendingFriends = () => ({
  type: FETCH_PENDING_FRIENDS
});

export const loadMorePendingFriends = () => ({
  type: LOAD_MORE_PENDING_FRIENDS
});

export const refreshPendingFriends = () => ({
  type: REFRESH_PENDING_FRIENDS
});

export const fetchGroupMembers = () => ({
  type: FETCH_GROUP_MEMBERS
});

export const loadMoreGroupMembers = () => ({
  type: LOAD_MORE_GROUP_MEMBERS
});

export const refreshGroupMembers = () => ({
  type: REFRESH_GROUP_MEMBERS
});
