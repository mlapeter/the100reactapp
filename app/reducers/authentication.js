import {
  FETCH_TOKEN,
  FETCH_TOKEN_RESULT,
  FETCH_TOKEN_ERROR,
  DECODE_TOKEN,
  DECODE_TOKEN_RESULT,
  DECODE_TOKEN_ERROR,
  REMOVE_TOKEN,
  REMOVE_TOKEN_ERROR,
  SET_FIREBASE_TOKEN
} from "../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TOKEN:
      return {
        ...state,
        username: action.username,
        password: action.password,
        isLoading: true
      };
    case FETCH_TOKEN_RESULT:
      return {
        ...state,
        token: action.token,
        firebaseToken: action.firebaseToken
      };
    case FETCH_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        token: "",
        firebaseToken: ""
      };
    case DECODE_TOKEN:
      return {
        ...state,
        token: action.token,
        isLoading: true
      };
    case DECODE_TOKEN_RESULT:
      return {
        ...state,
        user: action.result,
        isLoading: false,
        isAuthed: true
      };
    case DECODE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false
      };
    case REMOVE_TOKEN:
      return {
        ...state,
        isAuthed: false,
        isLoading: false,
        user: {},
        token: "",
        firebaseToken: ""
      };
    case REMOVE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        user: {},
        token: ""
      };
    case SET_FIREBASE_TOKEN:
      return {
        ...state,
        firebaseToken: action.token
      };
    default:
      return state;
  }
};
