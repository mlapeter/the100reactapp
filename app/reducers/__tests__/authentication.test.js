import reducer from "../authentication";
import { fetchToken } from "../../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

it("sets initial state", () => {
  const expected = { ...initialState };
  const actual = reducer(undefined, {});
  expect(actual).toEqual(expected);
});

it("sets initial state using snapshot", () => {
  expect(reducer(undefined, {})).toMatchSnapshot();
});

it("fetches token with valid username and password", () => {
  const expected = {
    ...initialState,
    username: "user1",
    password: "pw123",
    isLoading: true
  };
  const actual = reducer(undefined, fetchToken("user1", "pw123"));
  expect(actual).toEqual(expected);
});

it("fetches token with valid username and password using snapshot", () => {
  expect(reducer(undefined, fetchToken("user1", "pw123"))).toMatchSnapshot();
});
