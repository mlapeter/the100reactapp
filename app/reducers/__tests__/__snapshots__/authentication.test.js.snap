// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches token with valid username and password using snapshot 1`] = `
Object {
  "appLoading": true,
  "firebaseToken": "",
  "isAuthed": false,
  "isLoading": true,
  "password": "pw123",
  "token": "",
  "user": Object {},
  "username": "user1",
}
`;

exports[`sets initial state using snapshot 1`] = `
Object {
  "appLoading": true,
  "firebaseToken": "",
  "isAuthed": false,
  "isLoading": false,
  "token": "",
  "user": Object {},
}
`;
